pkg/apiserver/api_installer.go

```diff
@@ -321,10 +321,12 @@ func (a *APIInstaller) registerResourceHandlers(path string, storage rest.Storag
 
 			// list or post across namespace.
 			// TODO: more strongly type whether a resource allows these actions on "all namespaces" (bulk delete)
-			namer = scopeNaming{scope, a.group.Linker, gpath.Join(a.prefix, itemPath), true}
-			actions = appendIf(actions, action{"LIST", resource, params, namer}, isLister)
-			actions = appendIf(actions, action{"POST", resource, params, namer}, isCreater)
-			actions = appendIf(actions, action{"WATCHLIST", "watch/" + resource, params, namer}, allowWatchList)
+			if !hasSubresource {
```

Comment:
Hold is removed - if we need it we can easily fix it.

----- Original Message -----
> > @@ -321,10 +321,12 @@ func (a *APIInstaller) registerResourceHandlers(path
> > string, storage rest.Storag
> >  
> >  			// list or post across namespace.
> >  			// TODO: more strongly type whether a resource allows these actions on
> >  			"all namespaces" (bulk delete)
> > -			namer = scopeNaming{scope, a.group.Linker, gpath.Join(a.prefix,
> > itemPath), true}
> > -			actions = appendIf(actions, action{"LIST", resource, params, namer},
> > isLister)
> > -			actions = appendIf(actions, action{"POST", resource, params, namer},
> > isCreater)
> > -			actions = appendIf(actions, action{"WATCHLIST", "watch/" + resource,
> > params, namer}, allowWatchList)
> > +			if !hasSubresource {
> 
> @smarterclayton Do you still want the hold? Or is this fine to submit once it
> is LGTM'd?
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/GoogleCloudPlatform/kubernetes/pull/9991/files#r32762352